bytecomp/matching.ml

```diff
@@ -184,23 +379,16 @@ end = struct
 
   let combine ctx = List.map Row.combine ctx
 
-  let ctx_matcher p q rem =
-    let rec expand_record p =
-      match p.pat_desc with
-        | Tpat_record (l, _) ->
-           {p with pat_desc = Tpat_record (all_record_args l, Closed)}
-        | Tpat_alias (p, _, _) -> expand_record p
-        | _ -> p in
-    let ph, omegas =
-      let ph, p_args = Pattern_head.deconstruct (expand_record p) in
-      ph, List.map (fun _ -> omega) p_args in
-    let qh, args = Pattern_head.deconstruct (expand_record q) in
-    let yes () = (p, args @ rem) in
+  let ctx_matcher (ph : Pattern_head.t) (sq : Simple.pattern) rem =
+    let ph, sq = Pattern_head.expand_record ph, Simple.expand_record sq in
+    let arg_omegas = omegas (Pattern_head.arity ph) in
+    let qh, args = Pattern_head.deconstruct (Simple.to_pattern sq) in
```

Comment:
I think you could drop the `arity` function from `Pattern_head` and have
```ocaml
let arg_omegas = omega_list args
```

Reply:
Indeed, but I expect `arity` to be useful to get rid of the `OrPat` exception as well.